cli/internal/sbom/catalog.go

```diff
@@ -0,0 +1,49 @@
+package sbom
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/defenseunicorns/zarf/cli/internal/images"
+	"github.com/defenseunicorns/zarf/cli/internal/message"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/testifysec/witness/pkg/attestation"
+	"github.com/testifysec/witness/pkg/attestation/syft"
+)
+
+func CatalogImages(tagToImage map[name.Tag]v1.Image, sbomDir string) {
+	imageCount := len(tagToImage)
+	spinner := message.NewProgressSpinner("Creating SBOMs for %d images.", imageCount)
+	actx, err := attestation.NewContext([]attestation.Attestor{})
```

Comment:
as a pattern, we always run `defer spinner.Stop()` immediately after invoking `message.NewProgressSpinner()` to ensure there are no spinner leaks.

Reply:
Can make that change -- I copied the pattern I saw in images.PullAll

Reply:
Yeah we have some areas we may have missed that pattern, thx.